File: src/main/java/com/teamwizardry/wizardry/proxy/IProxy.java
package com.teamwizardry.wizardry.proxy;

import com.teamwizardry.wizardry.client.particle.GlitterBox;
import com.teamwizardry.wizardry.client.particle.KeyFramedGlitterBox;
import net.minecraft.item.ItemStack;
import net.minecraft.util.Hand;

public interface IProxy {

    void clientSetup();

    void registerHandlers();

    void setItemStackHandHandler(Hand hand, ItemStack stack);

    void spawnParticle(GlitterBox box);

    void spawnKeyedParticle(KeyFramedGlitterBox box);

    void openWorktableGui();
}

File: src/main/java/com/teamwizardry/wizardry/proxy/IProxy.kt
package com.teamwizardry.wizardry.proxy

import com.teamwizardry.wizardry.client.particle.GlitterBox
import net.minecraft.item.ItemStack
import net.minecraft.util.Hand

interface IProxy {
    fun clientSetup()
    fun registerHandlers()
    fun setItemStackHandHandler(hand: Hand?, stack: ItemStack?)
    fun spawnParticle(box: GlitterBox)

    //    void spawnKeyedParticle(KeyFramedGlitterBox box);
    fun openWorktableGui()
}